src/components/cylc/workflow/deltas.js

```diff
@@ -0,0 +1,130 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Vue from 'vue'
+import mergeWith from 'lodash/mergeWith'
+import isArray from 'lodash/isArray'
+import { mergeWithCustomizer } from '@/components/cylc/common/merge'
+
+/**
+ * @typedef {Object} Result
+ * @property {Array<Object>} errors
+ */
+
+/**
+ * @param {DeltasAdded} added
+ * @param {Object.<String, Object>} lookup
+ * @return {Result}
+ */
+function applyDeltasAdded (added, lookup) {
+  const result = {
+    errors: []
+  }
+  Object.values(added).forEach(addedValue => {
+    const items = isArray(addedValue) ? addedValue : [addedValue]
+    items.forEach(addedData => {
+      try {
+        Vue.set(lookup, addedData.id, addedData)
+      } catch (error) {
+        result.errors.push([
+          'Error applying added-delta, see browser console logs for more. Please reload your browser tab to retrieve the full flow state',
+          error,
+          addedData
+        ])
+      }
+    })
+  })
+  return result
+}
+
+/**
+ * Deltas updated.
+ *
+ * @param updated {DeltasUpdated} updated
+ * @param {Object.<String, Object>} lookup
+ * @return {Result}
+ */
+function applyDeltasUpdated (updated, lookup) {
+  const result = {
+    errors: []
+  }
+  Object.values(updated).forEach(updatedValue => {
+    const items = isArray(updatedValue) ? updatedValue : [updatedValue]
+    items.forEach(updatedData => {
+      try {
+        lookup[updatedData.id] = updatedData
+        const existingNode = lookup[updatedData.id]
+        if (existingNode) {
+          mergeWith(existingNode, updatedData, mergeWithCustomizer)
+        }
+      } catch (error) {
+        result.errors.push([
+          'Error applying updated-delta, see browser console logs for more. Please reload your browser tab to retrieve the full flow state',
+          error,
+          updatedData
+        ])
+      }
+    })
+  })
+  return result
+}
+
+/**
+ * Deltas pruned.
+ *
+ * @param {DeltasPruned} pruned - deltas pruned
+ * @param {Object.<String, Object>} lookup
+ * @return {Result}
+ */
+function applyDeltasPruned (pruned, lookup) {
+  Object.values(pruned).forEach(prunedData => {
+    for (const id of prunedData) {
+      if (lookup[id]) {
+        delete lookup[id]
+      }
+    }
+  })
+  return {
+    errors: []
+  }
+}
+
+/**
+ * @param {*} data
+ * @param {Object.<String, Object>} lookup
+ */
+export default function (data, lookup) {
+  const added = data.deltas.added
+  const updated = data.deltas.updated
+  const pruned = data.deltas.pruned
+  const errors = []
+  if (added) {
+    const result = applyDeltasAdded(added, lookup)
+    errors.push(...result.errors)
+  }
+  if (updated) {
+    const result = applyDeltasUpdated(updated, lookup)
+    errors.push(...result.errors)
+  }
+  if (pruned) {
+    const result = applyDeltasPruned(pruned, lookup)
+    errors.push(...result.errors)
+  }
+  return {
+    errors
+  }
+}
```

Comment:
This file contains the deltas for `workflow`. `workflows/workflow` is stored in Vuex, and used by GScan, Dashboard, and WorkflowTable component/views (for now, since the data structure is similar for all views/components, but that can be easily changed now).